demos/projects/ESPRESSIF/esp32/components/sample-azure-iot/transport_tls_esp32.c

```diff
@@ -38,6 +38,181 @@
+            function does that already
+        */      
+        /* TODO: Replace the below with your own implementation - the provided 
+            implementation is applicable to TFLX-PROTO devices only
+        */
+        *ppcRegistrationId = malloc(21);     
+        if(*ppcRegistrationId == NULL) {
+            return 3;
+        }
+        sprintf(*ppcRegistrationId,"sn%02X%02X%02X%02X%02X%02X%02X%02X%02X",pucHsmData[0],pucHsmData[1],\
+        pucHsmData[2],pucHsmData[3],pucHsmData[4],pucHsmData[5],pucHsmData[6],pucHsmData[7],pucHsmData[8]);
+        *(*ppcRegistrationId + 20) = '\0';
+
+        return 0;
+}
+
+#elif defined(CONFIG_ATECC608A_TCUSTOM)
+/**
+ * @brief [TrustCUSTOM] Dynamically generate and write the registration ID as a
+ *  string into the passed pointer
+ *
+ * @param[in,out] ppcRegistrationId Input: Pointer to a null pointer, 
+ *                      Output: Pointer to a null-terminated string
+ * @param[in]     pucHsmData Pointer to a buffer holding data to be passed
+ *                          (if any) to help generate the Registration ID
+ * @param[in]     ulHsmDataLength Length of the buffer passed in the 
+ *                          second parameter       
+ * 
+ * @return  0   if everything went through correctly 
+ */
+static uint32_t getRegistrationIdFromTCSM( char **ppcRegistrationId,\
+                                            uint8_t *pucHsmData,\
+                                            uint32_t ulHsmDataLength ) {
+
+        /* We don't check for NULL-ness of the input or the 
+            ability to talk to the HSM - the getRegistrationId(...)
+            function does that already
+        */      
+        /* TODO: Replace the below with your own implementation - the provided 
+            implementation is applicable to certs generated using esp-cryptoauth 
+            tool only
+        */
+        *ppcRegistrationId = malloc(19);     
+        if(*ppcRegistrationId == NULL) {
+            return 3;
+        }
+
+        sprintf(*ppcRegistrationId,"%02X%02X%02X%02X%02X%02X%02X%02X%02X",pucHsmData[0],pucHsmData[1],\
+        pucHsmData[2],pucHsmData[3],pucHsmData[4],pucHsmData[5],pucHsmData[6],pucHsmData[7],pucHsmData[8]);
+        *(*ppcRegistrationId + 18) = '\0';  
+        return 0;
+}
+#endif
+
+/**
+ * @brief Dynamically generate and write the registration ID as a
+ *  string into the passed pointer
+ *
+ * @param[in,out] ppcRegistrationId Input: Pointer to a null pointer, 
+ *                      Output: Pointer to a null-terminated string
+ * 
+ * @return  1  if the input is not a pointer to a NULL pointer,
+ *          2  if we are not able to talk to the HSM
+ *          3  if something else went wrong (eg: memory allocation failed)
+ *          0   if everything went through correctly 
+ */
+uint32_t getRegistrationId( char **ppcRegistrationId ) {
+
+        if(*ppcRegistrationId != NULL) {
+            return 1;
+        }
+        uint32_t ret = 0;
+        uint8_t sernum[9];
```

Comment:
same here

```c
#define tlsesp32SERIAL_NUMBER_SIZE 9
```

and then change this line to
```suggestion
        uint8_t sernum[tlsesp32SERIAL_NUMBER_SIZE];  
```